tests/phpunit/Presentation/ListFacetHtmlBuilderTest.php

```diff
@@ -106,9 +142,14 @@ public function testOnlyCheckboxesForValuesInThePropertyConstraintsAreChecked_or
 			typeSpecificConfig: [ 'defaultCombineWith' => 'OR' ]
 		);
 
-		$this->assertFalse( $viewModel['checkboxes'][0]['checked'] );
-		$this->assertTrue( $viewModel['checkboxes'][1]['checked'] );
-		$this->assertFalse( $viewModel['checkboxes'][2]['checked'] );
+		$this->assertFalse( $viewModel['checkboxes']['visible'][0]['checked'] );
+		$this->assertTrue( $viewModel['checkboxes']['visible'][1]['checked'] );
+		$this->assertFalse( $viewModel['checkboxes']['visible'][2]['checked'] );
+		$this->assertFalse( $viewModel['checkboxes']['visible'][3]['checked'] );
+		$this->assertFalse( $viewModel['checkboxes']['visible'][4]['checked'] );
+
+		$this->assertFalse( $viewModel['checkboxes']['collapsed'][0]['checked'] );
+		$this->assertFalse( $viewModel['checkboxes']['collapsed'][1]['checked'] );
```

Comment:
I see many existing tests got new entries and no new ones got added. There likely is an issue here. @malberts can you check?

Reply:
Some thoughts on the `StubValueCounter` change:
* It causes unnecessary/irrelevant assertions in other tests.
* Rather create a new test double, maybe `SequenceValueCounter`, that takes a `$count` int as a constructor parameter, and then generates `$count` amount of `ValueCount` in `countValues()`.
* Or else add that parameter in `StubValueCounter` and return only `$count` of the hardcoded values. But this adds some complication to the stub, so maybe it's better to keep it separate.

Regarding tests:
* add an explicit test for when the checkboxes count = boundary
* add an explicit test for when the count > boundary